onebusaway-gtfs/src/main/java/org/onebusaway/gtfs/model/BookingRule.java

```diff
@@ -0,0 +1,311 @@
+ * limitations under the License.
+ */
+package org.onebusaway.gtfs.model;
+
+import org.onebusaway.csv_entities.schema.annotations.CsvField;
+import org.onebusaway.csv_entities.schema.annotations.CsvFields;
+
+@CsvFields(filename = "booking_rules.txt", required = false)
+public final class BookingRule extends IdentityBean<AgencyAndId> {
+
+  public static final int MISSING_VALUE = -999;
+
+  @CsvField(name = "booking_rule_id", mapping = EntityFieldMappingFactory.class)
+  private String id;
+
+  /** booking_type
+   (Enum, Required) Defines how much in advance the booking can be made. Value are:
+   0: Real-time booking only
+   1: Up to same-day booking, with advance notice
+   2: Up to prior day(s) booking
+   */
+  @CsvField(name = "booking_type", mapping = EntityFieldMappingFactory.class)
+  private int type;
+
+  /**
+   prior_notice_duration_min
+   (Integer, Conditionally Required) Minimum number of minutes of advance time necessary before travel to make a booking request.
+
+   Conditionally Required:
+   (The timing must be provided by either a duration or a last time & day)
+   Required for up-to-same-day booking (booking_type=1)..
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_duration_min", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeDurationMin;
+
+  /**
+   prior_notice_duration_max
+   (Integer, Conditionally Optional) Maximum number of minutes of advance time necessary before travel to make a booking request.
+
+   Conditionally Optional:
+   Optional for up-to-same-day booking (booking_type=1)
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_duration_max", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeDurationMax;
+
+  /**
+   prior_notice_last_day
+   (Integer, Conditionally Required) Latest day on which a booking can be made. Defined as an offset, so the number of service days in advance of the booking.
+
+   Example: “Ride must be booked 1 day in advance before 5PM” will be encoded as prior_notice_last_day=1.
+
+   Conditionally Required:
+   (The timing must be provided by either a duration or a last time & day)
+   Required for up-to-prior-day booking (booking_type=2).
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_last_day", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeLastDay;
+
+  /**
+   prior_notice_last_time
+   (Time, Conditionally Required) Latest time of day on the last day on which a booking can be made. The timezone used is the one defined by agency.agency_timezone.
+
+   Example: “Ride must be booked 1 day in advance before 5PM” will be encoded as prior_notice_last_time=17:00:00.
+
+   Conditionally Required:
+   Required if prior_notice_last_day is defined.
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_last_time", optional = true, mapping = EntityFieldMappingFactory.class)
```

Comment:
use StopTimeFieldMappingFactory for the mapping

Reply:
I also removed the default values since they are not mentioned anywhere in the docs. I see that they are sometimes used for checking existence, but I am not particularly sure if we need it here too.